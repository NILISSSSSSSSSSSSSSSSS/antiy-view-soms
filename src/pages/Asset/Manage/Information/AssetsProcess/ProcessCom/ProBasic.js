import React from 'react'
import '../style.less'

export default ({ basicObj }) => {
  const { remarkName, note, fileInfo } = basicObj
  // 检查附件.doc
  const fileInfos = JSON.parse(fileInfo)
  const styles = {
    style: {
      display: 'block',
      overflow: 'hidden',
      'text-overflow': 'ellipsis',
      'white-space': 'nowrap'
    }
  }
  return (
    <section className="basic-info">
      <p>
        <span className="label">{remarkName}：</span>
        <span>{note}</span>
      </p>
      <p>
        <span className="label">附件：</span>
        {Array.isArray(fileInfos) && fileInfos.map((item, i)=> <a {...styles} onClick={() => window.open(`/api/v1/file/download?access_token=${sessionStorage.getItem('token')}&url=${item.url}&fileName=${encodeURIComponent(item.fileName)}`)}>{item.fileName}</a> )}
      </p>
    </section>
  )
}

