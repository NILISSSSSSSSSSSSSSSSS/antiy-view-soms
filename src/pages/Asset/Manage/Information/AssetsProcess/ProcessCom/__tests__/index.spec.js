import React from 'react'
import ProBasic from '../ProBasic'
import { shallow } from 'enzyme'

const fileInfo = [{
  'url': '666',
  'fileName': '666'
}]
const props = {
  basicObj: {
    remarkName: 666,
    note: 666,
    fileInfo: JSON.stringify(fileInfo)
  }
}

describe('<ProBasic />', () => {
  const wrapper = shallow(<ProBasic {...props}/>)

  it('render <AssetsSearch /> component', () => {
    expect(wrapper.find('.basic-info')).toHaveLength(1)
  })
  it('test <a></a> click',  ()=> {
    wrapper.find('a').simulate('click')
  })

  it('shoulder render a ProBasic',  ()=> {
    const app = wrapper.debug()
    expect(app).toMatchSnapshot()
  })

})