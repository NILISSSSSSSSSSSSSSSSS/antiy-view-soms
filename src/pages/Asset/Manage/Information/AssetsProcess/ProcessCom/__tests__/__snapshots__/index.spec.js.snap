// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProBasic /> shoulder render a ProBasic 1`] = `
"<section className=\\"basic-info\\">
  <p>
    <span className=\\"label\\">
      666
      ：
    </span>
    <span>
      666
    </span>
  </p>
  <p>
    <span className=\\"label\\">
      附件：
    </span>
    <a onClick={[Function: onClick]}>
      666
    </a>
  </p>
</section>"
`;
